src/internal/modules/billing/controller.js

```diff
@@ -0,0 +1,176 @@
+const { selectBillingTypeForm, billingTypeFormSchema } = require('./forms/billing-type');
+const { selectBillingRegionForm, billingRegionFormSchema } = require('./forms/billing-region');
+const { viewBillRunListForm } = require('./forms/bill-run-list');
+const services = require('internal/lib/connectors/services');
+const forms = require('shared/lib/forms');
+const { get } = require('lodash');
+const moment = require('moment');
+const queryString = require('querystring');
+const helpers = require('@envage/water-abstraction-helpers');
+
+/**
+ * Step 1a of create billing batch flow - display form to select type
+ * i.e. Annual, Supplementary, Two-Part Tariff
+ * @param {*} request
+ * @param {*} h
+ */
+const getBillingBatchType = async (request, h) => {
+  const sessionForm = request.yar.get(get(request, 'query.form'));
+  if (sessionForm) {
+    request.yar.clear(get(request, 'query.form'));
+  }
+  return h.view('nunjucks/form', {
+    ...request.view,
+    back: '/manage',
+    form: sessionForm || selectBillingTypeForm(request)
+  });
+};
+
+const getBillingRegions = async () => {
+  const { data } = await services.water.billingBatchCreateService.getBillingRegions();
+  return data;
+};
+
+/**
+ * Step 1b - receive posted step 1a data
+ * @param {*} request
+ * @param {*} h
+ */
+const postBillingBatchType = async (request, h) => {
+  const billingTypeForm = forms.handleRequest(selectBillingTypeForm(request), request, billingTypeFormSchema(request));
+
+  if (billingTypeForm.isValid) {
+    const { selectedBillingType } = forms.getValues(billingTypeForm);
+    return h.redirect(`/billing/batch/region/${selectedBillingType}`);
+  }
+
+  const key = uuid();
+  request.yar.set(key, billingTypeForm);
+  return h.redirect('/billing/batch/type?' + queryString.stringify({ form: key }));
+};
+
+/**
+ * Step 2a - display selelct region form
+ * @param {*} request
+ * @param {*} h
+ */
+const getBillingBatchRegion = async (request, h) => {
+  const sessionForm = request.yar.get(get(request, 'query.form'));
+  if (sessionForm) {
+    request.view.form = sessionForm;
+    request.yar.clear(get(request, 'query.form'));
+  }
+
+  const regions = await getBillingRegions();
+
+  return h.view('nunjucks/form', {
+    ...request.view,
+    back: '/billing/batch/type',
+    form: sessionForm || selectBillingRegionForm(request, regions)
+  });
+};
+
+const getBatchDetails = (billingRegionForm, userEmail) => {
+  const { selectedBillingType, selectedBillingRegion } = forms.getValues(billingRegionForm);
+  const financialYear = (new Date().getMonth > 3) ? helpers.charging.getFinancialYear() + 1 : helpers.charging.getFinancialYear();
+  const batch = {
+    'userEmail': userEmail,
+    'regionId': selectedBillingRegion,
+    'batchType': selectedBillingType,
+    'financialYearEnding': financialYear,
+    'season': 'all year' // ('summer', 'winter', 'all year').required();
+  };
+  return batch;
+};
+
+const getUserEmail = async (request) => {
```

Comment:
I don't think we need this. The email should already be on the request at `request.defra.userName` or `request.defra.user.user_name`

Reply:
thanks - done